mongodbatlas/data_source_mongodbatlas_search_index.go

```diff
@@ -0,0 +1,72 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceMongoDBAtlasSearchIndex() *schema.Resource {
+	return &schema.Resource{
+		Read:   dataSourceMongoDBAtlasSearchIndexRead,
+		Schema: returnSearchIndexSchema(),
+	}
+
+}
+
+func dataSourceMongoDBAtlasSearchIndexRead(d *schema.ResourceData, meta interface{}) error {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID, projectIDOk := d.GetOk("project_id")
+	clusterName, clusterNameOK := d.GetOk("cluster_name")
+	indexID, indexIDOk := d.GetOk("index_id")
+
+	if !projectIDOk || !clusterNameOK || !indexIDOk {
```

Comment:
we can probably simplify this to `!(projectIDOk || clusterNameOK || indexIDOk)` or if we are checking that all of those arguments are set use `&&`

Reply:
on my way